packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/checkfortoken.md

```diff
@@ -1,36 +1,42 @@
-Check for an existing session in two steps.
+To check for an existing session may take two steps:
 
-1. Initialize the authorization state manager (`authStateManager`) from a saved session.
-2. Verify that the access token is still valid by checking the expiry.
+- Check for an existing valid token.
+- Refresh the token if it has expired.
 
-The state manager session and its associated token are already saved by the `signIn` function calling `writeToSecureStorage`.
+The `Credential` class manages the tokens for a user. The `signIn()` function you wrote in [Open the sign-in page](#open-the-sign-in-page) calls `store(_:)` to save the users' credentials. A session exists if there's a default credential. Although there are calls to check if a credential has expired and to request a refresh, this code uses `refreshIfNeeded()` that only tries to refresh the token if it's expired.
```

Comment:
```suggestion
The `Credential` class manages the tokens for a user. The `signIn()` function you wrote in [Open the sign-in page](#open-the-sign-in-page) calls `store(_:)` to save the user's credentials. A session exists if there's a default credential. Although there are calls to check if a credential has expired and to request a refresh, this code uses `refreshIfNeeded()` that only tries to refresh the token if it's expired.
```

Reply:
Nice catch "There can be only one." –Highlander. Change is in the next commit.